[DataGrid] Build error: "Cannot find module '../../..'" and "Cannot find namespace 'Chai'"

<!-- Provide a general summary of the issue in the Title above -->
I tried to use the Data-grid component (which is super awesome by the way!).
I installed it from npmjs (`npm install --save @material-ui/data-grid`) and import it in my project. I discovered some build errors, corresponding to module import.
<!--
  Thank you very much for contributing to Material-UI X by creating an issue! ❤️
  To avoid duplicate issues we ask you to check off the following list.
-->

<!-- Checked checkbox should look like this: [x] -->

- [ X] The issue is present in the latest release.
- [ X] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

<!-- Describe what happens instead of the expected behavior. -->
I use the data-grid component in my project with the following code :
```
[...]
import { DataGrid, ColDef } from '@material-ui/data-grid';
[...]
const DATAGRID_COLUMNS: ColDef[] = [
    { field: 'id', headerName: 'ID', width: 250 },
    { field: 'name', headerName: 'Nom', width: 300 },
    {
        field: 'latitude',
        headerName: 'Latitude',
        width: 130,
    },
    {
        field: 'longitude',
        headerName: 'Longitude',
        width: 130,
    },
]
[...]
return (
[...]
 <DataGrid columns={DATAGRID_COLUMNS} rows={stations} pageSize={10} />
[...]
);
```
I run the `npm run build` and the following errors occured : 
```
ERROR in [at-loader] ./node_modules/@material-ui/data-grid/dist/data-grid.d.ts:1645:96 
    TS2307: Cannot find module '../../..' or its corresponding type declarations.

ERROR in [at-loader] ./node_modules/@material-ui/data-grid/dist/data-grid.d.ts:1794:13 
    TS2503: Cannot find namespace 'Chai'.
```

Despite these errors, the build finished and the rendering is exactly what I excepted.

## Expected Behavior 🤔

<!-- Describe what should happen. -->
I expected to have the exact same render but without any errors.

## Steps to Reproduce 🕹

<!--
  Provide a link to a live example (you can use codesandbox.io) and an unambiguous set of steps to reproduce this bug.
  Include code to reproduce, if relevant (which it most likely is).

  You should use the official codesandbox template as a starting point: https://material-ui.com/r/issue-template

  If you have an issue concerning TypeScript please start from this TypeScript playground: https://material-ui.com/r/ts-issue-template

  Issues without some form of live example have a longer response time.
-->

Steps:

1. install the data-grid component
2. import it in my project
3. run the build script from npm
4. discover errors during build

## Context 🔦

<!--
  What are you trying to accomplish? How has this issue affected you?
  Providing context helps us come up with a solution that is most useful in the real world.
-->
These errors during the build is not a good thing since I build this project in a CI and it does not allow errors during builds.


## Your Environment 🌎

<!--
  Include as many relevant details about the environment with which you experienced the bug.
  If you encounter issues with TypeScript please include version and tsconfig.
-->


tsconfig.json : 
```
{
    "compilerOptions": {
        "outDir": "./cubicweb_tdev_realtime/data/",
        "sourceMap": true,
        "noImplicitAny": true,
        "declaration": true,
        "strictNullChecks": true,
        "module": "CommonJS",
        "target": "ES6",
        "jsx": "react"
    },
    "include": ["./cubicweb_appjs/appjs/**/*"]
}
```


| Tech          | Version |
| ------------- | ------- |
| Material-UI X | v4.0.0-alpha.9  |
| React         |     16.13.1    |
| Browser       |      Firefox 78.4.1esr   |
| TypeScript    |      4.0.5   |
| npm          |   6.14.8      |

Thank you for your quick answer!

This fix the second error, which is : 
```
ERROR in [at-loader] ./node_modules/@material-ui/data-grid/dist/data-grid.d.ts:1794:13 
    TS2503: Cannot find namespace 'Chai'.
```

To be honest I don't really understand why but it works :).

But the first error is still there : 
```
ERROR in [at-loader] ./node_modules/@material-ui/data-grid/dist/data-grid.d.ts:1645:96 
    TS2307: Cannot find module '../../..' or its corresponding type declarations.
```

It seems to be related to the `reselect` (https://github.com/reduxjs/reselect) lib, which is used here https://github.com/mui-org/material-ui-x/blob/b7cb4f36368487a9b0bfb3befdeaa25708b52453/packages/grid/_modules_/grid/hooks/features/selection/selectionSelector.ts#L6
This leads to the following code in `packages/grid/data-grid/dist/data-grid.d.ts` : 
```
declare const selectedRowsCountSelector: OutputSelector<GridState, number, (res: Record<import("../../..").RowId, boolean>) => number>
```
And the `"../../.."` import is not suitable here.

Nevertheless, I do not know how to fix this...

--- a/packages/grid/_modules_/grid/hooks/features/selection/selectionSelector.ts
+++ b/packages/grid/_modules_/grid/hooks/features/selection/selectionSelector.ts
@@ -1,9 +1,13 @@
-import { createSelector } from 'reselect';
+import { createSelector, OutputSelector } from 'reselect';
 import { GridState } from '../core/gridState';
 import { SelectionState } from './selectionState';
 
 export const selectionStateSelector = (state: GridState) => state.selection;
-export const selectedRowsCountSelector = createSelector<GridState, SelectionState, number>(
+export const selectedRowsCountSelector: OutputSelector<
+  GridState,
+  number,
+  (res: SelectionState) => number
+> = createSelector<GridState, SelectionState, number>(
   selectionStateSelector,
   (selection) => Object.keys(selection).length,
 );

--- a/packages/grid/tsconfig.build.json
+++ b/packages/grid/tsconfig.build.json
@@ -10,5 +10,5 @@
     "rootDir": "./"
   },
   "include": ["./data-grid/src", "./x-grid/src", "./_modules_/**/*"],
-  "exclude": ["__tests__", "**/*.test.ts", "node_modules"]
+  "exclude": ["__tests__", "**/*.test.*", "node_modules"]
 }

